### PR TITLE
Move Python 3.10 testing to 3.11

### PR DESCRIPTION
**Pull request by jgraham**
To match the pattern of testing the latest and oldest supported versions.

### CODE REVIEW

**Comment by jgraham** [comment]
I can reproduce the failure locally. It basically seems like there's [no aioquic binary wheel](https://pypi.org/project/aioquic/0.9.20/#files) for 3.11. So I don't think we can update before that's released.

**Comment by foolip** [comment]
@jgraham I compared to https://github.com/web-platform-tests/wpt/pull/40192 and push a commit with changes you'd missed. I started from https://github.com/web-platform-tests/wpt/pull/33706 and made sure I had changes to the same files, and looked into any differences.

**Comment by foolip** [comment]
@jgraham is https://github.com/aiortc/aioquic/issues/372 the only blocker you're aware of here?

**Comment by jlaine** [comment]
aioquic 0.9.21 is now on PyPI with wheels for Python 3.7 and up!

File: tools/pytest.ini [hunk 1]
```diff
@@ -28,3 +28,7 @@ filterwarnings =
     always:the imp module is deprecated in favour of importlib:DeprecationWarning
     # https://github.com/web-platform-tests/wpt/issues/39827
     always:pkg_resources is deprecated as an API:DeprecationWarning
+    # mozdevice uses pipes module
+    ignore:'pipes' is deprecated and slated for removal in Python 3.13:DeprecationWarning
+    # mozrunner uses telnetlib module
+    ignore:'telnetlib' is deprecated and slated for removal in Python 3.13:DeprecationWarning
```
Comment (gsnedders) [inline tools/pytest.ini:34]:
```suggestion
    # mozdevice uses pipes module
    ignore:'pipes' is deprecated and slated for removal in Python 3:DeprecationWarning
    # mozrunner uses telnetlib module
    ignore:'telnetlib' is deprecated and slated for removal in Python 3:DeprecationWarning
```

Let's avoid being overly specific, given deprecations are often pushed back a release or two.

But also move this up to all the ignores above, leaving the things we care about at the bottom?

**Review by thiagowfx — APPROVED** [review]

**Review by gsnedders — APPROVED** [review]

**Pull request merged**